src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -213,30 +219,80 @@ public boolean isOnOrderPieceStatus(CheckInPiece checkinPiece) {
    *
    * @param compPOL   PO line to retrieve/create Item Records for
    * @param holdingId holding uuid from the inventory
-   * @param expectedQuantity expected items quantity for the holding
+   * @param locations list of locations holdingId is associated with
    * @return future with list of item id's
    */
-  private CompletableFuture<List<String>> handleItemRecords(CompositePoLine compPOL, String holdingId, int expectedQuantity) {
+  private CompletableFuture<List<Piece>> handleItemRecords(CompositePoLine compPOL, String holdingId, List<Location> locations) {
+    Map<Piece.Format, Integer> piecesWithItems = calculatePiecesQuantity(compPOL, locations, true);
+    int piecesWithItemsQty = IntStreamEx.of(piecesWithItems.values()).sum();
+    String polId = compPOL.getId();
+
+    logger.debug("Handling {} items for PO Line with id={} and holdings with id={}", piecesWithItemsQty, polId, holdingId);
```

Comment:
I'm not going to be a stickler about this sort of thing, but generally anything logged at a level below INFO should be conditionalized... i.e. only executed if the logger is configured to write messages of that log level.

Reply:
This way is actually recommended by [Log4j 2 API](https://logging.apache.org/log4j/2.0/manual/api.html#Substituting_Parameters) so logging framework will construct log message if proper logger level is enabled.